src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/PhysicalPlanWithCost.java

```diff
@@ -22,4 +24,28 @@ public double getWeight() {
         return this.cost;
     }
 
+    public static PhysicalPlanWithCost findPlanWithSmallestCost( final List<PhysicalPlanWithCost> plansWithCost ) {
+        PhysicalPlanWithCost bestPlan = plansWithCost.get(0);
+        double min = bestPlan.getWeight();
+
+        for ( int i = 1; i < plansWithCost.size(); i++) {
+            final PhysicalPlanWithCost p = plansWithCost.get(i);
+            if ( p.getWeight() <= min ) {
+                min = p.getWeight();
+                bestPlan = p;
+            }
+        }
+        return bestPlan;
+    }
+
+    public static double calculateAvgCostOfPlans( final List<PhysicalPlanWithCost> plansWithCost ) {
+        double sum = 0;
+
+        for ( final PhysicalPlanWithCost p : plansWithCost ) {
+            sum += p.getWeight();
+        }
+
+        return sum/plansWithCost.size();
+    }
```

Comment:
My mistake. I wanted to ask you to create a new class called `PhysicalPlanWithCostUtils`. Can you please do that. Thanks! ...and sorry for causing you double work here.